### ISSUE TITLE
Repeated NullPointerExceptions

### ISSUE THREAD

**Issue by dtaylor84** [ubergeek42/weechat-android#302]
I can't quite figure out what is causing these, but once one happens it will continue to happen immediately upon re-connecting for a period of time.  Then all will be well again (for a while), before the cycle repeats.

logcat shows:

```
09-10 21:02:07.106 26881 26902 E AndroidRuntime: FATAL EXCEPTION: r0
09-10 21:02:07.106 26881 26902 E AndroidRuntime: Process: com.ubergeek42.WeechatAndroid, PID: 26881
09-10 21:02:07.106 26881 26902 E AndroidRuntime: java.lang.NullPointerException: Attempt to invoke virtual method 'java.lang.String[] java.lang.String.split(java.lang.String)' on a null object reference
09-10 21:02:07.106 26881 26902 E AndroidRuntime:        at com.ubergeek42.weechat.relay.protocol.Data.getHdata(Data.java:173)
09-10 21:02:07.106 26881 26902 E AndroidRuntime:        at com.ubergeek42.weechat.relay.protocol.Data.getObject(Data.java:273)
09-10 21:02:07.106 26881 26902 E AndroidRuntime:        at com.ubergeek42.weechat.relay.RelayMessage.<init>(RelayMessage.java:2238)
09-10 21:02:07.106 26881 26902 E AndroidRuntime:        at com.ubergeek42.weechat.relay.connection.AbstractConnection.access$300(AbstractConnection.java:2147)
09-10 21:02:07.106 26881 26902 E AndroidRuntime:        at com.ubergeek42.weechat.relay.connection.AbstractConnection$2.run(AbstractConnection.java:83)
09-10 21:02:07.106 26881 26902 E AndroidRuntime:        at java.lang.Thread.run(Thread.java:761)
09-10 21:02:07.108 26881 26902 I Process : Sending signal. PID: 26881 SIG: 9
09-10 21:02:07.144   892   903 I ActivityManager: Process com.ubergeek42.WeechatAndroid (pid 26881) has died
09-10 21:02:07.144   892   903 W ActivityManager: Service crashed 2 times, stopping: ServiceRecord{9e5faf7 u0 com.ubergeek42.WeechatAndroid/.service.RelayService}
```
WeechatAndroid 0.12-git build ID v0.11-83-g9426987
WeeChat 1.6-rc1 (git: v1.6-rc1-15-gce17b2e) [compiled on Sep  7 2016 23:06:39]

**Comment by oakkitten** [comment]
apparently a slight change in the protocol has been causing this. should be fixed in 8cb241a45db7270b675f19b75467839b0cf69dc4 on master

**Comment by dtaylor84** [comment]
That definitely seems much more stable, thanks!
